### PR TITLE
OnnxBackendNodeModelTest XFAILs cleanup

### PR DESCRIPTION
**Pull request by Ewa21**

### CODE REVIEW

**Review by bsledz — CHANGES_REQUESTED** [review]

File: ngraph/python/tests/test_onnx/test_backend.py [hunk 1]
```diff
@@ -272,8 +264,6 @@ def expect_fail(test_case_path, xfail):  # type: (str) -> None
     (xfail_issue_38706,
      "OnnxBackendNodeModelTest.test_split_zero_size_splits_cpu"),
     (xfail_issue_38736,
-     "OnnxBackendNodeModelTest.test_softmax_cross_entropy_sum_log_prob_expanded_cpu",
-     "OnnxBackendNodeModelTest.test_softmax_cross_entropy_input_shape_is_NCd1d2d3d4d5_none_no_weight_log_prob_expanded_cpu",  # noqa
      "OnnxBackendNodeModelTest.test_softmax_cross_entropy_sum_expanded_cpu",
      "OnnxBackendNodeModelTest.test_softmax_cross_entropy_input_shape_is_NCd1d2d3_sum_weight_high_ignore_index_expanded_cpu",  # noqa
      "OnnxBackendNodeModelTest.test_softmax_cross_entropy_mean_no_weight_ignore_index_3d_log_prob_expanded_cpu",  # noqa
```
[the remark this answers is not in the document]
Comment (Ewa21) [reply ngraph/python/tests/test_onnx/test_backend.py]: Done

**Review by mitruska — APPROVED** [review]

File: ngraph/python/tests/test_onnx/test_backend.py [hunk 2]
```diff
@@ -173,24 +169,17 @@ def expect_fail(test_case_path, xfail):  # type: (str) -> None
...
     (xfail_issue_39662,
      "OnnxBackendNodeModelTest.test_nonmaxsuppression_two_classes_cpu",
-     "OnnxBackendNodeModelTest.test_slice_default_axes_cpu",
      "OnnxBackendNodeModelTest.test_scatter_elements_with_negative_indices_cpu",
-     "OnnxBackendNodeModelTest.test_constant_pad_cpu",
-     "OnnxBackendNodeModelTest.test_edge_pad_cpu",
-     "OnnxBackendNodeModelTest.test_reflect_pad_cpu",
      "OnnxBackendNodeModelTest.test_top_k_cpu",
      "OnnxBackendNodeModelTest.test_top_k_negative_axis_cpu",
      "OnnxBackendNodeModelTest.test_top_k_smallest_cpu",
```
Comment (mitruska) [inline ngraph/python/tests/test_onnx/test_backend.py:175]: Please double check issue label for TopK

Comment (Ewa21) [reply ngraph/python/tests/test_onnx/test_backend.py]: Done

File: ngraph/python/tests/test_onnx/test_backend.py [hunk 3]
```diff
@@ -132,15 +129,15 @@ def expect_fail(test_case_path, xfail):  # type: (str) -> None
 globals().update(backend_test.enable_report().test_cases)
 
 tests_expected_to_fail = [
-    (xfail_issue_34314,
+    (xfail_issue_49207,
```
Comment (rblaczkowski) [inline ngraph/python/tests/test_onnx/test_backend.py:132]: First appearance of xfail_issue_49207

File: ngraph/python/tests/test_onnx/test_backend.py [hunk 4]
```diff
@@ -132,15 +129,15 @@ def expect_fail(test_case_path, xfail):  # type: (str) -> None
...
      "OnnxBackendNodeModelTest.test_rnn_seq_length_cpu",
      "OnnxBackendNodeModelTest.test_simple_rnn_defaults_cpu",
      "OnnxBackendNodeModelTest.test_simple_rnn_with_initial_bias_cpu"),
-    (xfail_issue_33540,
+    (xfail_issue_49207,
```
Comment (rblaczkowski) [inline ngraph/python/tests/test_onnx/test_backend.py:136]: Second appearance of: xfail_issue_49207

File: ngraph/python/tests/test_onnx/test_backend.py [hunk 5]
```diff
@@ -173,24 +170,18 @@ def expect_fail(test_case_path, xfail):  # type: (str) -> None
...
+     "OnnxBackendNodeModelTest.test_top_k_smallest_cpu"),
     (xfail_issue_33633,
      "OnnxBackendNodeModelTest.test_maxpool_2d_dilations_cpu"),
-    (xfail_issue_38086,
-     "OnnxBackendNodeModelTest.test_dynamicquantizelinear_min_adjusted_expanded_cpu",
-     "OnnxBackendNodeModelTest.test_dynamicquantizelinear_expanded_cpu",
-     "OnnxBackendNodeModelTest.test_dynamicquantizelinear_max_adjusted_expanded_cpu",
-     "OnnxBackendNodeModelTest.test_quantizelinear_cpu"),
-    (xfail_issue_34310,
+    (xfail_issue_49207,
```
Comment (rblaczkowski) [inline ngraph/python/tests/test_onnx/test_backend.py:184]: third appearance of: xfail_issue_49207

Comment (rblaczkowski) [reply ngraph/python/tests/test_onnx/test_backend.py:184]: 3 time of using xfail_issue_49207 in separated blocks - I suppose that this is not intentional

**Review by rblaczkowski — CHANGES_REQUESTED** [review]

File: ngraph/python/tests/test_onnx/test_backend.py (hunk 5 above)
Comment (Ewa21) [reply ngraph/python/tests/test_onnx/test_backend.py]: Thank you for pointing it out :)

**Review by bsledz — APPROVED** [review]

**Review by rblaczkowski — APPROVED** [review]

**Review by postrational — APPROVED** [review]

**Pull request merged**